web/frontend/src/types/configuration.ts

```diff
@@ -0,0 +1,49 @@
+type ID = string;
+
+// Text describes a "text" question, which allows the user to enter free text.
+interface Text {
```

Comment:
It would be better to have a common interface for questions:

```ts
// Question defines a common interface for questions.
interface Question {
  ID: ID;
  Title: string;
  MaxN: number;
  MinN: number;
  Type: string;
}

// Text describes a "text" question, which allows the user to enter free text.
interface Text extends Question {
  MaxLength: number;
  Regex?: string;
  Choices: string[];
}
```